### PR TITLE
Issue 161: Create external access guide

### LINKED ISSUE #161: Documenting Serviceaccount, roles & clusterroles required for enabling external-connectivity for Pravega
**When enabling `external-connectivity` if the service account that is configured in the SegmentStore pod doesn't have proper permissions then we will encounter issue https://github.com/pravega/pravega/issues/3614 .** 

So it would be good to have a proper configuration (Serviceaccount, roles, clusterroles, etc ..) details documented in `README.md` page https://github.com/pravega/pravega-operator#enable-external-access.

As per the suggestion from @adrianmo, below are the steps to be followed for enabling `external-connectivity` 

1. Create a service account
```
apiVersion: v1
kind: ServiceAccount
metadata:
  name: pravega-components
```
2. Create a role (make sure to use the correct namespace) 
```
kind: Role
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: pravega-components
  namespace: "default"
rules:
- apiGroups: ["pravega.pravega.io"]
  resources: ["*"]
  verbs: ["get"]
- apiGroups: [""]
  resources: ["pods", "services"]
  verbs: ["get"]
```

3. Create the clusterrole
```
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: pravega-components
rules:
- apiGroups: [""]
  resources: ["nodes"]
  verbs: ["get"]
```
4. Create rolebinding
```
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: pravega-components
subjects:
- kind: ServiceAccount
  name: pravega-components
roleRef:
  kind: Role
  name: pravega-components
  apiGroup: rbac.authorization.k8s.io
```
5. Create clusterrolebinding
```
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: pravega-components
subjects:
- kind: ServiceAccount
  name: pravega-components
  namespace: default
roleRef:
  kind: ClusterRole
  name: pravega-components
  apiGroup: rbac.authorization.k8s.io
```
6. On the `PravegaCluster` manifest, configure the service accounts
```
apiVersion: "pravega.pravega.io/v1alpha1"
kind: "PravegaCluster"
metadata:
  name: "example"
spec:
  bookkeeper:
    serviceAccountName: pravega-components
...
  pravega:
    controllerServiceAccountName: pravega-components
    segmentStoreServiceAccountName: pravega-components
...
```

### PR DESCRIPTION
**Pull request by adrianmo**
### Change log description
- Create external access guide with instructions to configure it, enable it and sample manifests.

### Purpose of the change
- Fix #161 

---
Signed-off-by: Adrián Moreno <adrian@morenomartinez.com>

### CODE REVIEW

**Comment by adrianmo** [comment]
@deenav @Tristan1900 can you have a look at this? Thanks.

File: doc/external-access.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# Enable external access
+
+By default, a Pravega cluster uses `ClusterIP` services which are only accessible from within Kubernetes. However, when creating the Pravega cluster resource, you can opt to enable external access.
+
+In Pravega, clients initiate the communication with the Pravega Controller, which is a stateless component frontended by a Kubernetes service that load-balances the requests to the backend pods. Then, clients discover the individual Segment Store instances to which they directly read and write data to. Clients need to be able to reach each and every Segment Store pod in the Pravega cluster.
+
+If your Pravega cluster needs to be consumed by clients from outside Kubernetes (or from another Kubernetes deployment), you can enable external access in two ways, depending on your environment constraints and requirements. Both ways will create one service for all Controllers, and one service for each Segment Store pod.
+
+1. Via [`LoadBalancer`](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer) service type.
+2. Via [`NodePort`](https://kubernetes.io/docs/concepts/services-networking/service/#nodeport) service type.
```
Comment (Tristan1900) [inline doc/external-access.md]: Do you think we need to explicitly mention that `NodePort` is not supported in PKS+NSX-T environment? Thanks.

Comment (adrianmo) [reply doc/external-access.md]: No, I don't think so. This operator is not dependent on any particular Kubernetes platform (e.g. GKE, EKS, AKS, PKS...). We shouldn't attempt to enumerate them and keep track of the features they support and the features they don't, especially considering that that's an information that will be changing over time. It's up to the users to investigate what service type is suitable for them in terms of support and use case.

Comment (Tristan1900) [reply doc/external-access.md]: Make sense. Thanks!

**Pull request merged**